tests/virtual_machines_test.go

```diff
@@ -532,6 +532,14 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					_, err = net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
 					Expect(err).ToNot(HaveOccurred(), "should successfully parse assigned mac")
 				})
+				It("should remove the mac from vmWaitConfigMap after vm created successfully", func() {
+					By("Checking vmWaitConfigMap is cleared")
+					Eventually(func() map[string]string {
+						data, err := getVmWaitConfigMapData()
+						Expect(err).ToNot(HaveOccurred(), "Should succeed getting vmWaitConfigMap")
+						return data
+					}, timeout, pollingInterval).Should(BeEmpty(), "vmWaitConfigMap should be empty after successful vm creation")
+				})
```

Comment:
I think you need to move vm creation to `BeforeEach` for this to work

Reply:
totally right.